src/annotator/integrations/test/pdf-test.js

```diff
@@ -369,15 +373,119 @@ describe('PDFIntegration', () => {
   describe('#scrollToAnchor', () => {
     it('scrolls to first highlight of anchor', async () => {
       const highlight = document.createElement('div');
-      document.body.appendChild(highlight);
+      const offset = 42;
+      const integration = createPDFIntegration();
+      fakeScrollUtils.offsetRelativeTo
+        .withArgs(highlight, integration.contentContainer())
+        .returns(offset);
 
       const anchor = { highlights: [highlight] };
+      await integration.scrollToAnchor(anchor);
 
+      assert.calledOnce(fakeScrollUtils.scrollElement);
+      assert.calledWith(
+        fakeScrollUtils.scrollElement,
+        integration.contentContainer(),
+        offset
+      );
+    });
+
+    it('does not scroll if anchor has no highlights', async () => {
       const integration = createPDFIntegration();
+      const anchor = {};
+
       await integration.scrollToAnchor(anchor);
 
-      assert.calledOnce(fakeScrollIntoView);
-      assert.calledWith(fakeScrollIntoView, highlight, sinon.match.func);
+      assert.notCalled(fakeScrollUtils.scrollElement);
+    });
+
+    /**
+     * Create an anchor whose highlight is inside a placeholder for a non-rendered
+     * PDF page.
+     */
+    function createPlaceholderHighlight() {
+      const container = document.createElement('div');
+      const placeholder = createPlaceholder(container);
+      const highlight = document.createElement('div');
+      placeholder.append(highlight);
+      return highlight;
+    }
+
+    it('waits for anchors in placeholders to be re-anchored and scrolls to final highlight', async () => {
+      const placeholderHighlight = createPlaceholderHighlight();
+      const integration = createPDFIntegration();
+      fakeScrollUtils.offsetRelativeTo
+        .withArgs(placeholderHighlight, integration.contentContainer())
+        .returns(50);
+      const annotation = { $tag: 'tag1' };
+      const anchor = { annotation, highlights: [placeholderHighlight] };
+
+      // Check that the PDF content was scrolled to the approximate position of
+      // the anchor, indicated by the placeholder.
+      const scrollDone = integration.scrollToAnchor(anchor);
+      assert.calledWith(
+        fakeScrollUtils.scrollElement,
+        integration.contentContainer(),
+        50
+      );
+
+      // Simulate a delay while rendering of the text layer for the page happens
+      // and re-anchoring completes.
+      await delay(5);
+
+      // Create a new anchor for the annotation created by re-anchoring.
+      const finalHighlight = document.createElement('div');
+      fakeScrollUtils.scrollElement.resetHistory();
+      fakeAnnotator.anchors.push({
```

Comment:
It's opaque here what happens when you push an `anchor` object to the `annotator`'s `anchors` collection that pertains to an `annotation` already referenced by a (or multiple) `anchor` objects in that collection. Given that `Array.prototype.find` (used in the method under test) will only return the first match, it matters. Can you help shed some light on this?

Reply:
In practice all annotations created with Hypothesis or stored in h will only ever have a single target and thus one anchor. If an annotation happened to have multiple targets (eg. because it had been stored in some non-h backend that supported it) then only the first would be used.